src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml/Given_FrameworkElement_And_Leak.cs

```diff
@@ -0,0 +1,94 @@
+﻿#if !WINDOWS_UWP
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Private.Infrastructure;
+using Uno.UI.RuntimeTests.Tests.Windows_UI_Xaml.Controls;
+using Windows.UI.Core;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace Uno.UI.RuntimeTests.Tests.Windows_UI_Xaml
+{
+	[TestClass]
+	[RunsOnUIThread]
+	public class Given_FrameworkElement_And_Leak
+	{
+		private static ConditionalWeakTable<FrameworkElement, Holder> _holders
+			= new ConditionalWeakTable<FrameworkElement, Holder>();
+
+		[TestMethod]
+		[DataRow(typeof(XamlEvent_Leak_UserControl), 15)]
+		[DataRow(typeof(XamlEvent_Leak_UserControl_xBind), 15)]
+		[DataRow(typeof(XamlEvent_Leak_UserControl_xBind_Event), 15)]
+		public async Task When_Add_Remove(Type controlType, int count)
+		{
+			var maxCounter = 0;
+			var activeControls = 0;
+			var maxActiveControls = 0;
+
+			var rootContainer = new ContentControl();
+
+			TestServices.WindowHelper.WindowContent = rootContainer;
+
+			await TestServices.WindowHelper.WaitForIdle();
+
+			for (int i = 0; i < count; i++)
+			{
+				var item = Activator.CreateInstance(controlType) as FrameworkElement;
+				_holders.Add(item, new Holder(HolderUpdate));
+				rootContainer.Content = item;
+				await TestServices.WindowHelper.WaitForIdle();
+				rootContainer.Content = null;
+				GC.Collect();
+				GC.WaitForPendingFinalizers();
+				await Task.Yield();
+			}
+
+			void HolderUpdate(int value)
+			{
+				var unused = rootContainer.Dispatcher.RunAsync(CoreDispatcherPriority.High,
```

Comment:
Codacy found an issue: [Remove the unused local variable 'unused'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6112846)